beancount/core/data.py

```diff
@@ -34,9 +33,8 @@
 Meta = Dict[str, Any]
 
 
-@runtime_checkable
 class BeancountError(Protocol):
-    """Beancount errors are classes with these attributes."""
+    """Beancount errors are namedtuple with these attributes"""
```

Comment:
With this protocol, we don't need them to be namedtuples, right? (so e.g. plugins are free to use plain classes or dataclasses). If you don't like the word "classes", maybe just "objects"?